resharper/resharper-unity/src/CSharp/Daemon/UsageChecking/UsageInspectionsSuppressor.cs

```diff
@@ -92,13 +121,83 @@ public bool SuppressUsageInspectionsOnElement(IDeclaredElement element, out Impl
                     // But it still should be checked if the field is ever accessed from the code.
                     flags = ImplicitUseKindFlags.Assign;
                     return true;
-                
+
+                case IProperty property when IsImplicitlyUsedInterfaceProperty(property):
+                    flags = ImplicitUseKindFlags.Assign;
+                    return true;
+
                 case IProperty property when IsEventHandler(unityApi, property.Setter):
                     flags = ImplicitUseKindFlags.Assign;
                     return true;
             }
 
-            flags = ImplicitUseKindFlags.Default;   // Value not used if we return false
+            flags = ImplicitUseKindFlags.Default; // Value not used if we return false
+            return false;
+        }
+
+        private bool IsImplicitlyUsedInterfaceMethod(IMethod method)
+        {
+            foreach (var implicitlyUsedTypeName in myImplicitlyUsedInterfaces)
+            {
+                var type = TypeFactory.CreateTypeByCLRName(implicitlyUsedTypeName, method.Module).GetTypeElement();
+
+                if (type == null)
+                    return false;
+                
+                foreach (var overridableMemberInstance in method.GetRootSuperMembers())
+                {
+                    if (type.DerivesFrom(overridableMemberInstance.DeclaringType.GetClrName()))
```

Comment:
Does `DeclaringType` always return back the interface? If so, we can simplify this to just comparing type names, and skipping the method name check too.

We'll need to check that `DeclaringType` is always the interface, even in the case of a derived class that is overriding a method declared in a base class that implements one of our interesting interfaces.

Reply:
Why wouldn't we need to check the method name?

Reply:
We know that any method/property on these interfaces should be marked as implicitly used, so if the `overridableMemberInstance.DeclaringType` is the interface itself (and not a derived type), then the method has to belong to that interface, and will be implicitly used.